api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java

```diff
@@ -9,58 +9,110 @@
+	 * @see UserByNameComparator#compare(User, User)
 	 */
 	@Test
 	public void compare_shouldSortUsersByPersonNames() {
-		
-		Person person1 = new Person();
-		person1.addName(new PersonName("givenName", "middleName", "familyName"));
-		User user1 = new User(person1);
-		Person person2 = new Person();
-		person2.addName(new PersonName("givenName", "middleNamf", "familyName"));
-		User user2 = new User(person2);
+
 		Person person3 = new Person();
 		person3.addName(new PersonName("givenName", "middleNamg", "familyName"));
 		User user3 = new User(person3);
 		Person person4 = new Person();
 		person4.addName(new PersonName("givenName", "middleNamh", "familyName"));
 		User user4 = new User(person4);
-		
+
 		List<User> listToSort = new ArrayList<>();
 		// add the users randomly
 		listToSort.add(user3);
 		listToSort.add(user1);
 		listToSort.add(user4);
 		listToSort.add(user2);
-		
+
 		// sort the list with userByNameComparator
 		listToSort.sort(new UserByNameComparator());
-		
+
 		// make sure that the users are sorted in the expected order
 		Iterator<User> it = listToSort.iterator();
 		Assert.assertTrue("Expected user1 to be the first in the sorted user list but wasn't", user1.equals(it.next()));
 		Assert.assertTrue("Expected user2 to be the second in the sorted user list but wasn't", user2.equals(it.next()));
 		Assert.assertTrue("Expected user3 to be the third in the sorted user list but wasn't", user3.equals(it.next()));
 		Assert.assertTrue("Expected user4 to be the fourth in the sorted user list but wasn't", user4.equals(it.next()));
 	}
+
+	/**
+	 * This tests sorting with the {@link UserByNameComparator} given a set of users with
+	 * personNames (first parameter null and second parameter as valid user)
+	 *
+	 * @see UserByNameComparator#compare(User, User)
+	 */
+	@Test
+	public void compare_shouldReturnUser2FirstAfterBeingSortedIfUser1IsNull() {
+		List<User> listToSort = new ArrayList<>();
+		// added the users randomly
+		listToSort.add(nullUser);
+		listToSort.add(user2);
+
+		listToSort.sort(new UserByNameComparator());
+		Iterator<User> sortedListIterator = listToSort.iterator();
+
+		Assert.assertTrue("Expected user2 to be the first in the sorted user list but wasn't",
+			user2.equals(sortedListIterator.next()));
+
+	}
+
+	/**
+	 * This tests sorting with the {@link UserByNameComparator} given a set of users with
+	 * personNames (first parameter as valid user and second parameter as null)
+	 *
+	 * @see UserByNameComparator#compare(User, User)
+	 */
+	@Test
+	public void compare_shouldReturnUser1FirstAfterBeingSortedIfUser2IsNull() {
+
+		List<User> listToSort = new ArrayList<>();
+		// added the users randomly
+		listToSort.add(nullUser);
+		listToSort.add(user1);
+
+		listToSort.sort(new UserByNameComparator());
+		Iterator<User> sortedListIterator = listToSort.iterator();
+
+		Assert.assertTrue("Expected user1 to be the first in the sorted user list but wasn't",
+			user1.equals(sortedListIterator.next()));
+	}
```

Comment:
What if there is a list having multiple null users? Testing this for a list having just 2 users drops this case. A more robust test can make use of atleast 2 users and 2 null users.

Reply:
okay... makes sense.. i'll try it.. 👍 
let me know if there are more such things that need to be worked on..  :)

Reply:
done.. please check once..

Reply:
i wanted to make sure there's no error when it comes to null-null (consecutive) comparison..... but now I realised, that condition would get tested somewhere in the alternate null test too.. :/ 
sorry for the confusion.. should i remove the consecutive one and push the code? 
is there anything else to be covered?

Reply:
Yes, remove that test.

Reply:
Yes remove the test